pandas/tests/indexing/test_indexing.py

```diff
@@ -627,6 +627,13 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    def test_empty_index_for_empty_dataframe(self):
+        index=pd.Index([], name='idx')
+        df = pd.DataFrame(columns=['A'], index=index)
+        df2 = df.copy()
+        df['A'] = []
```

Comment:
use 
result=
expected=
tm.assert_index_equal

but actually, pls construct the expected DataFrame and compare with the result frame.